.github/workflows/pr-env-deploy.yml

```diff
@@ -332,3 +332,53 @@ jobs:
 
       - name: Cleanup Environment Test Data
         run: make test-performance-cleanup TF_WORKSPACE_NAME=${{ needs.set-environment-id.outputs.environment_id }}
+
+  smoke-test:
+    name: Run Smoke Tests
+    needs: [set-environment-id, deploy]
+    environment: pull-request
+    runs-on: [self-hosted, ci]
+
+    steps:
+      - name: Git Clone - ${{ github.event.pull_request.head.ref }}
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Setup asdf cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.asdf
+          key: ${{ runner.os }}-asdf-${{ hashFiles('**/.tool-versions') }}
+          restore-keys: |
+            ${{ runner.os }}-asdf-
+
+      - name: Install asdf and tools
+        uses: asdf-vm/actions/install@v3.0.2
+        with:
+          asdf_branch: v0.13.1
+
+      - name: Python Dependency Install
+        run: poetry install --no-root
+
+      - name: Configure Management Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-2
+          role-to-assume: ${{ secrets.MGMT_ROLE_ARN }}
+          role-session-name: github-actions-ci-${{ needs.set-environment-id.outputs.environment_id }}
+
+      - name: Retrieve Client Certificates
+        run: make truststore-pull-client ENV=dev
+
+      - name: Configure Dev Account Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-2
+          role-chaining: true
+          role-to-assume: ${{ secrets.DEPLOY_ROLE_ARN }}
+          role-session-name: github-actions-ci-${{ needs.set-environment-id.outputs.environment_id }}
+
+      - name: Smoke Test
+        run: |
+          make TF_WORKSPACE_NAME=${{ needs.set-environment-id.outputs.environment_id }} test-smoke-public
```

Comment:
this won't run against the PR env - `test-smoke-public` uses the public endpoint which will be `internal-dev`. I think you'll want `test-smoke-internal` instead.